common/v2/components/AccountList.tsx

```diff
@@ -192,10 +193,26 @@ export default function AccountList(props: AccountListProps) {
     privacyCheckboxEnabled = false,
     dashboard
   } = props;
-  const { deleteAccountFromCache } = useContext(StoreContext);
+  const { deleteAccountFromCache, restoreDeletedAccount, accountRestore } = useContext(
+    StoreContext
+  );
   const { updateAccount } = useContext(AccountContext);
-  const [deletingIndex, setDeletingIndex] = useState();
-  const overlayRows = [deletingIndex];
+  const [deletingIndex, setDeletingIndex] = useState<number | undefined>();
+  const [undoDeletingIndexes, setUndoDeletingIndexes] = useState<[number, TUuid][]>([]);
+  const overlayRows: [number[], [number, TUuid][]] = [
+    isNumber(deletingIndex) ? [deletingIndex] : [],
+    [...undoDeletingIndexes]
+  ];
+
+  const getDisplayAccounts = (): StoreAccount[] => {
+    const accountsTemp = cloneDeep(displayAccounts);
```

Comment:
The "issue" with spread operators is that they don't deep copy the object. So I wanted to play it safe, and create a hard copy, since I am deleting them.